### ISSUE TITLE
(🐞) Mypy doesn't use `python_executable` to find plugins

### ISSUE THREAD

**Issue by KotlinIsland** [python/mypy#12575]
```toml
[tool.mypy]
python_executable = ".venv/bin/python"
plugins = ["pydantic.mypy"]
```

```py
# test.py
import pydantic
```
```
(.venv) > mypy test.py
Success: no issues found in 1 source file
(.venv) > deactivate
...
> mypy test.py
pyproject.toml:1: error: Error importing plugin "pydantic.mypy": No module named 'pydantic'
Found 1 error in 1 file (errors prevented further checking)
```

**Comment by baco** [comment]
Same here:
```
$ mypy --python-executable ../../.venv/bin/python erp/tasks.py
pyproject.toml:1:1: error: Error importing plugin "mypy_django_plugin.main": No module named
'mypy_django_plugin'  [misc]
    # This file's location could be a level above, outside the Django site dir, but
    ^
Found 1 error in 1 file (errors prevented further checking)
```

**Comment by pranavrajpal** [comment]
I think this is by design. mypy needs to be able to import all the plugins specified (since mypy needs to be able to call functions in that plugin and plugins probably want to be able to use functions in mypy itself) and it can't really do that if the plugin is installed with a different Python than mypy.

**Comment by hauntsaninja** [comment]
Yeah, Pranav is right. If you're using plugins, you really want mypy to be installed in the target environment
